parfive/tests/test_downloader.py

```diff
@@ -343,3 +349,58 @@ def test_ftp_http(tmpdir, httpserver):
     f = dl.download()
     assert len(f) == 2
     assert len(f.errors) == 4
+
+
+def test_default_user_agent(event_loop, httpserver, tmpdir):
+    tmpdir = str(tmpdir)
+    httpserver.serve_content('SIMPLE  = T',
+                             headers={'Content-Disposition': "attachment; filename=testfile.fits"})
+
+    dl = Downloader(loop=event_loop)
+    dl.enqueue_file(httpserver.url, path=Path(tmpdir), max_splits=None)
+
+    assert dl.queued_downloads == 1
+
+    dl.download()
+
+    assert 'User-Agent' in httpserver.requests[0].headers
+    assert httpserver.requests[0].headers['User-Agent'] == f"parfive/{parfive.__version__} aiohttp/{aiohttp.__version__} python/{sys.version[:5]}"
+
+
+def test_custom_user_agent(event_loop, httpserver, tmpdir):
+    tmpdir = str(tmpdir)
+    httpserver.serve_content('SIMPLE  = T',
+                             headers={'Content-Disposition': "attachment; filename=testfile.fits"})
+
+    dl = Downloader(loop=event_loop, headers={'User-Agent': 'test value 299792458'})
+    dl.enqueue_file(httpserver.url, path=Path(tmpdir), max_splits=None)
+
+    assert dl.queued_downloads == 1
+
+    dl.download()
+
+    assert 'User-Agent' in httpserver.requests[0].headers
+    assert httpserver.requests[0].headers['User-Agent'] == "test value 299792458"
+
+
+@patch.dict(os.environ,{'HTTP_PROXY': "proxy_url"})
+def test_proxy_passed_as_kwargs_to_get(event_loop, tmpdir):
```

Comment:
You also want to test a https url now. I suggest you use https://docs.pytest.org/en/latest/example/parametrize.html to pass in two URLs and expected proxy url.

You should also set a different proxy url for http and https to ensure it's using the right one.

Reply:
Does this work?

Reply:
yeah looks good.